components/script/dom/htmlinputelement.rs

```diff
@@ -2728,18 +2728,18 @@ impl Activatable for HTMLInputElement {
             // https://html.spec.whatwg.org/multipage/#reset-button-state-%28type=reset%29:activation-behaviour-2
             // https://html.spec.whatwg.org/multipage/#checkbox-state-%28type=checkbox%29:activation-behaviour-2
             // https://html.spec.whatwg.org/multipage/#radio-button-state-%28type=radio%29:activation-behaviour-2
-            InputType::Submit |
-            InputType::Reset |
-            InputType::File |
-            InputType::Checkbox |
-            InputType::Radio => self.is_mutable(),
+            InputType::Submit | InputType::Reset | InputType::File => self.is_mutable(),
+            InputType::Checkbox | InputType::Radio => true,
             _ => false,
         }
     }
 
     // https://dom.spec.whatwg.org/#eventtarget-legacy-pre-activation-behavior
     fn legacy_pre_activation_behavior(&self) -> Option<InputActivationState> {
-        if !self.is_mutable() {
+        if !self.is_mutable() &&
+            self.input_type() != InputType::Checkbox &&
+            self.input_type() != InputType::Radio
+        {
```

Comment:
Based on https://html.spec.whatwg.org/multipage/input.html#the-input-element:legacy-pre-activation-behavior it looks like this entire check can be removed instead.